src/server/server/frontendstructs.py

```diff
@@ -103,19 +103,21 @@ class UserComputedProfile(object):
     Represent the information about the user profile:
     """
     @staticmethod
-    def make_from_scratch(feature_vector, model_data):
-        return UserComputedProfile(feature_vector, model_data, datetime=None)
+    def make_from_scratch(initial_feature_vector, feature_vector, model_data):
+        return UserComputedProfile(initial_feature_vector, feature_vector, model_data, datetime=None)
 
     @staticmethod
-    def make_from_db(feature_vector, model_data, datetime):
-        return UserComputedProfile(feature_vector, model_data, datetime)
+    def make_from_db(initial_feature_vector, feature_vector, model_data, datetime):
+        return UserComputedProfile(initial_feature_vector, feature_vector, model_data, datetime)
 
```

Comment:
I think this "initial_feature_vector" should be in struct.UserProfileModelData. it's not a clear cut, but I see this as a intermediate step to compute the only ONE feature_vector that should be used (and exposed) to match with documents. Here I find it ambiguous which one should be used in which case, we mix at the same level one of the 3 vectors used to compute the "final" one with this "final" vector, while the other 2 are elsewhere